pkg/controllermanager/controller/project/project_control.go

```diff
@@ -109,10 +109,10 @@ func (r *projectReconciler) Reconcile(ctx context.Context, request reconcile.Req
 	projectLogger.Infof("[PROJECT RECONCILE] %s", project.Name)
 
 	if project.DeletionTimestamp != nil {
-		return r.delete(ctx, project, r.gardenClient.Client(), r.gardenClient.Client())
+		return r.delete(ctx, project, r.gardenClient.Client(), r.gardenClient.APIReader())
 	}
 
-	return r.reconcile(ctx, project, r.gardenClient.Client(), r.gardenClient.Client())
+	return r.reconcile(ctx, project, r.gardenClient.Client(), r.gardenClient.APIReader())
```

Comment:
Ufff, this will put a lot of load on the API server and etcd after a restart (e.g. rolling update).
Every project is read with quorum read from etcd...

Reply:
This is reasonable.
`APIReader()` was changed to `APIClient()` in #4111. I am not sure that was it changed because of the issue you described? Regarding reverting the PR we can consider other's opinions also. Meanwhile, I will try to figure out new logic for project reconciliation.